stores/slabbuffer.go

```diff
@@ -307,10 +307,9 @@ func (mgr *SlabBufferManager) SlabBuffers() (sbs []api.SlabBuffer) {
 
 func (mgr *SlabBufferManager) SlabsForUpload(ctx context.Context, lockingDuration time.Duration, minShards, totalShards uint8, set uint, limit int) (slabs []api.PackedSlab, _ error) {
 	mgr.mu.Lock()
-	buffers := mgr.completeBuffers[bufferGID(minShards, totalShards, uint32(set))]
```

Comment:
I think a deep copy of the buffers should also work. The main lock shouldn't be held while performing disk i/o if avoidable.

Reply:
I figured we couldn't do that because we do lock checks for one (`lockedUntil`) but we also have to copy a pointer to the file which might get closed by another process while we're reading from it. Right?

Reply:
Ok, did the deep copy and run it in a loop and no race. Probably fine for now since this'll get removed with RHPv4